src/middlewares/auth.middleware.js

```diff
@@ -0,0 +1,31 @@
+import { usersService } from '../services/users.service.js';
+import { authService } from '../services/auth.service.js';
+import { ApiError } from '../exceptions/api.error.js';
+
+export const authMiddleware = async (req, res, next) => {
+  const { authorization } = req.headers;
+
+  if (!authorization) {
+    throw ApiError.Unauthorized();
+  }
+
+  const [, token] = authorization.split(' ');
+
+  if (!token) {
+    throw ApiError.Unauthorized();
+  }
+
+  const tokenData = authService.verifyAccessToken(token);
```

Comment:
The method 'verifyAccessToken' is likely asynchronous if it involves any I/O operations, such as database access or network requests. Consider using 'await' to ensure the token verification completes before proceeding.